### PR TITLE
Refactor cloud-services centric args

### LINKED ISSUE #76: Change -a <application> -t <test-type> to:  -d <--sub-directory to prefs>
Currently we use a cloud-services centric way to specify a path to prefs.
-a loop-server -t e2e-test would tell ff-tool to look for prefs in the following dir:
./loop-server/e2e-test/prefs.ini

However, we could make the tool more universal with something like:
-d loop-server/e2e-test
which would stipulate a relative path to your prefs file.  there is nothing else in ff-tool specifically tied to either test-type or application.

This would also preclude users from having to clone services-test (or use exact same file structure) in order to load their own custom browser prefs

### PR DESCRIPTION
**Pull request by rpappalax**
r?  @pdehaan

Fixes #76

### CODE REVIEW

File: fftool/arg_parser.py [hunk 1]
```diff
@@ -61,6 +49,12 @@ def arg_parser():
     )
 
     parser.add_argument(
```
Comment (rpappalax) [inline fftool/arg_parser.py]: Don't know if this is overkill, but when wifi goes down, we coul probably 'silently' check for that and proceed, but then the user might be thinking they're getting the latest browser (which isn't cool).  Might be better to let them fail, but then offer them the option to do --no-download (and run with cached).  Heck, it worked well on the plane :)

File: fftool/firefox_profile.py [hunk 2]
```diff
@@ -106,8 +117,8 @@ def create_mozprofile(profile_dir, application=None, test_type=None, env=None):
 
     prefs = Preferences()
 
-    for path in prefs_paths(application, test_type, env):
-        # print('PREFS.ADD_FILE(PATH): ' + path)
+    for path in prefs_paths(prefs_dirs):
```
Comment (rpappalax) [inline fftool/firefox_profile.py]: here we reduce 3 args into 1

File: README.rst [hunk 3]
```diff
@@ -198,5 +198,6 @@ new one with that name.
 
 ::
 
-   $ ff -c beta -p my_cool_profile1 -a loop-server -t e2e-test -f stage
+   $ ff -c beta -p my_cool_profile1 -d loop-server/e2e-test:stage
+   $ ff -c nightly -p my_cool_profile2 -d shavar/e2e-test:stage+moztestpub
```
Comment (pdehaan) [inline README.rst]: May need to add a section in the README somewhere for the new `--no-download` flag.

**Comment by pdehaan** [comment]
Not sure if its a config issue, or a lack of me specifying params, but this implodes pretty hard:

```bash
$ ff --no-download

================================================================================
FF-TOOL: download, install & launch Firefox!
================================================================================

Traceback (most recent call last):
  File "/Users/pdehaan/dev/github/ff-tool/venv/bin/ff", line 9, in <module>
    load_entry_point('ff-tool', 'console_scripts', 'ff')()
  File "/Users/pdehaan/dev/github/ff-tool/fftool/main.py", line 44, in main
    prefs_dirs=options.prefs_dirs,
  File "/Users/pdehaan/dev/github/ff-tool/fftool/firefox_profile.py", line 120, in create_mozprofile
    for path in prefs_paths(prefs_dirs):
  File "/Users/pdehaan/dev/github/ff-tool/fftool/firefox_profile.py", line 60, in prefs_paths
    for prefs_dir in prefs_dirs:
TypeError: 'NoneType' object is not iterable
```

File: README.rst (hunk 3 above)
Comment (rpappalax) [inline README.rst]: offline use could be:
1. wifi is down, i have FF cached, use cache
2. wifi is down, i don' t have FF cached, what now??
3. wifi is up, but i want to use cache anyway, use cache

Comment (pdehaan) [inline README.rst]: > 2\. wifi is down, i don' t have FF cached, what now??

Not sure how we detect if wifi is up or down (if we care), but my assumption here is that...
If `--no-download` is `True`, check to see if the specified/default channel's binary is cached.
  - If `True`, launch with the cached binary. Easy peasy.
  - If `False`, just fail or print some message saying "could not find cached Firefox binary at {binary path}".

**Comment by pdehaan** [comment]
Apart from my two smallish nits, I think this is fantastic! 🎉

**Comment by rpappalax** [comment]
awesome man, thanks!

**Comment by rpappalax** [comment]
@pdehaan: the only thing I didn't address in this PR is your notes on os.path.  dhunt explained to me that if we're already using os.path.join this will already create cross-platform compatible paths for us.  Using: os.path.normpath(path), abspath, etc. will do things like expand a path into users home dir or flesh out abs path, etc.  Seems like there's some fanciness there we could eventually explore.
It might be useful to revisit our whole path structure at some point, but that seems like a separate enhancement so I'll leave it off here.

Thx for all the notes. let me know if anything else that could use enfixination or OK to enmerginate.

**Comment by rpappalax** [comment]
I believe this also Closes #85.

It's not often that the download link fails, but if an HTTPS ConnectionError happens, we now should be handling that.

**Comment by pdehaan** [comment]
Beautiful. R+!

**Pull request merged**